includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -0,0 +1,399 @@
+<?php
+/**
+ * WC_Admin_Reports_Customers_Data_Store class file.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Reports_Customers_Data_Store.
+ */
+class WC_Admin_Reports_Customers_Data_Store extends WC_Admin_Reports_Data_Store implements WC_Admin_Reports_Data_Store_Interface {
+
+	/**
+	 * Table used to get the data.
+	 *
+	 * @var string
+	 */
+	const TABLE_NAME = 'wc_customer_lookup';
+
+	/**
+	 * Mapping columns to data type to return correct response types.
+	 *
+	 * @var array
+	 */
+	protected $column_types = array(
+		'customer_id'      => 'intval',
+		'user_id'          => 'intval',
+		'date_registered'  => 'strval',
+		'date_last_active' => 'strval',
+		'orders_count'     => 'intval',
+		'total_spend'      => 'floatval',
+		'avg_order_value'  => 'floatval',
+	);
+
+	/**
+	 * SQL columns to select in the db query and their mapping to SQL code.
+	 *
+	 * @var array
+	 */
+	protected $report_columns = array(
+		'customer_id'      => 'customer_id',
+		'user_id'          => 'user_id',
+		'username'         => 'username',
+		'name'             => "CONCAT_WS( ' ', first_name, last_name ) as name", // TODO: what does this mean for RTL?
+		'email'            => 'email',
+		'country'          => 'country',
+		'city'             => 'city',
+		'postcode'         => 'postcode',
+		'date_registered'  => 'date_registered',
+		'date_last_active' => 'date_last_active',
+		'orders_count'     => 'orders_count',
+		'total_spend'      => 'total_spend',
+		'avg_order_value'  => 'avg_order_value',
+	);
+
+	/**
+	 * Set up all the hooks for maintaining and populating table data.
+	 */
+	public static function init() {
+		add_action( 'woocommerce_new_order', array( __CLASS__, 'sync_order' ) );
+		add_action( 'woocommerce_update_order', array( __CLASS__, 'sync_order' ) );
+		add_action( 'woocommerce_order_refunded', array( __CLASS__, 'sync_order' ) );
+	}
+
+	/**
+	 * Maps ordering specified by the user to columns in the database/fields in the data.
+	 *
+	 * @param string $order_by Sorting criterion.
+	 * @return string
+	 */
+	protected function normalize_order_by( $order_by ) {
+		if ( 'name' === $order_by ) {
+			return "CONCAT_WS( ' ', first_name, last_name )";
```

Comment:
Disregard my question above about ordering. The only thing I can see this causing problems is some people might anticipate ordering by name to be last name driven.

Reply:
Should we make a change to that?